src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenGenerator.cs

```diff
@@ -66,7 +66,12 @@ public AntiforgeryToken GenerateCookieToken()
             if (identity != null && identity.IsAuthenticated)
```

Comment:
I looked at the code. It does not check for `IsAuthenticated`. It simply selects first non-null identity. https://github.com/dotnet/corefx/blob/master/src/System.Security.Claims/src/System/Security/Claims/ClaimsPrincipal.cs